addon/services/scroller.js

```diff
@@ -36,12 +38,19 @@ export default Em.Service.extend({
   },
 
   scrollVertical (target, opts = {}) {
-    this.get('scrollable').animate({
-      scrollTop: this.get('scrollable').scrollTop() - this.get('scrollable').offset().top + this.getVerticalCoord(target, opts.offset)
-    },
-      opts.duration || this.get('duration'),
-      opts.easing   || this.get('easing'),
-      opts.complete
-    );
+    return new RSVP.Promise((resolve) => {
+      this.get('scrollable').animate({
+          scrollTop: this.get('scrollable').scrollTop() - this.get('scrollable').offset().top + this.getVerticalCoord(target, opts.offset)
+        },
+        opts.duration || this.get('duration'),
+        opts.easing   || this.get('easing'),
+        function() {
+          if (typeof opts.complete === 'function') {
+            opts.complete.apply(this, arguments);
+          }
+          resolve();
```

Comment:
Did not think about the possibility of the animation being able to fail, but you are certainly right, if jQuery provides that callback we should support that. Just pushed the change!